dali/operators/decoder/audio/audio_decoder_impl.cc

```diff
@@ -17,39 +17,62 @@
 
 namespace dali {
 
-template <typename T>
-span<char> as_raw_span(T *buffer, ptrdiff_t length) {
-  return make_span(reinterpret_cast<char*>(buffer), length*sizeof(T));
+std::pair<int64_t, int64_t> ProcessOffsetAndLength(const AudioMetadata &meta, float offset_sec, float length_sec) {
+  int64_t offset = 0;
+  int64_t length = meta.length;
+  if (offset_sec > 0.0f)
+    offset = static_cast<int64_t>(offset_sec * meta.sample_rate);
+  if (length_sec > 0.0f)
+    length = static_cast<int64_t>(length_sec * meta.sample_rate);
+
+  // Limit the duration to the bounds of the input
+  if ((offset + length) > meta.length) {
+    length = meta.length - offset;
+  }
+  return {offset, length};
 }
 
 TensorShape<> DecodedAudioShape(const AudioMetadata &meta, float target_sample_rate,
-                                bool downmix) {
+                                bool downmix, float offset_sec, float duration_sec) {
   bool should_resample = target_sample_rate > 0 && meta.sample_rate != target_sample_rate;
   bool should_downmix = meta.channels > 1 && downmix;
   int channels = meta.channels == 1 || downmix ? 1 : meta.channels;
-  int64_t len = should_resample ? kernels::signal::resampling::resampled_length(
-                                      meta.length, meta.sample_rate, target_sample_rate)
-                                : meta.length;
+
+  int64_t offset, length;
+  std::tie(offset, length) = ProcessOffsetAndLength(meta, offset_sec, duration_sec);
+
+  int64_t len = should_resample ?
+    kernels::signal::resampling::resampled_length(length, meta.sample_rate, target_sample_rate) : length;
   return downmix ? TensorShape<>{len} : TensorShape<>{len, channels};
 }
 
-template <typename T, typename DecoderOutputType>
+template <typename T>
 void DecodeAudio(TensorView<StorageCPU, T, DynamicDimensions> audio, AudioDecoderBase &decoder,
                  const AudioMetadata &meta, kernels::signal::resampling::Resampler &resampler,
-                 span<DecoderOutputType> decode_scratch_mem,
+                 span<float> decode_scratch_mem,
                  span<float> resample_scratch_mem,
                  float target_sample_rate, bool downmix,
-                 const char *audio_filepath) {  // audio_filepath for debug purposes
+                 const char *audio_filepath,  // audio_filepath for debug purposes
+                 float offset_sec, float duration_sec) {
```

Comment:
```suggestion
                 double offset_sec, double duration_sec) {
```

Reply:
This doesn't seem like a good API - the offset and duration must have already been processed in order to allocate appropriate `audio` tensor - passing them here, especially in seconds, can only lead to inconsistency.
Suggested alternatives:
- `seek` in the decoder before this call and read `audio.shape[0]` samples
- pass the offset in samples and read `audio.shape[0]` samples.